node/pkg/fetcher/fetcher_test.go

```diff
@@ -334,6 +338,60 @@ func TestFetcherFetch(t *testing.T) {
 	}
 }
 
+func TestFetcherFetchProxy(t *testing.T) {
+	ctx := context.Background()
+	clean, testItems, err := setup(ctx)
+	if err != nil {
+		t.Fatalf("error setting up test: %v", err)
+	}
+	defer clean()
+
+	fetcher := testItems.fetcher
+
+	proxyServer := goproxy.NewProxyHttpServer()
+	proxyServer.Verbose = true
+	srv := &http.Server{
+		Addr:    ":8088",
+		Handler: proxyServer,
+	}
+	go func() {
+		if proxyServeErr := srv.ListenAndServe(); proxyServeErr != http.ErrServerClosed {
+			// Unexpected server shutdown
+			log.Fatal().Err(proxyServeErr).Msg("unexpected server shutdown")
+		}
+	}()
+
+	proxy, err := tests.PostRequest[Proxy](testItems.app, "/api/v1/proxy", map[string]any{"protocol": "http", "host": "localhost", "port": 8088})
+	if err != nil {
+		t.Fatalf("error creating proxy: %v", err)
+	}
+
+	err = fetcher.initialize(ctx)
+	if err != nil {
+		t.Fatalf("error initializing fetcher: %v", err)
+	}
+
+	for _, adapter := range fetcher.Adapters {
+		result, err := fetcher.fetch(*adapter)
+		if err != nil {
+			t.Fatalf("error fetching: %v", err)
+		}
+		assert.Greater(t, len(result), 0)
+	}
+
+	defer func() {
```

Comment:
@nick-bisonai I think I miss the point of using `defer` here. Wouldn't it be exactly the same as is we take the code outside of `defer func`?

Reply:
yes, you're correct, I'll remove defer func block